src/coap_client_zephyr.c

```diff
@@ -391,15 +391,19 @@ static int add_observation(golioth_coap_request_msg_t *req, struct golioth_clien
         return GOLIOTH_ERR_QUEUE_FULL;
     }
 
-    int err = golioth_coap_observe(req, client);
+    /* Store request message in Golioth client */
+    memcpy(&obs_info->req, req, sizeof(obs_info->req));
```

Comment:
Interesting, we should discuss the details. I think it would involve changing the program flow of `coap_io_lop_once` which is where the request is allocated and populated from the mailbox.

https://github.com/golioth/golioth-firmware-sdk/blob/f5202d8688c083f45c22680fdd35dbf9aa4ed86d/src/coap_client_zephyr.c#L414-L426